src/flp/test_types.rs

```diff
@@ -0,0 +1,112 @@
+use crate::field::FieldElement;
+use crate::flp::{FlpError, Gadget, Type};
+use std::convert::{TryFrom, TryInto};
+
+/// The average type. Each measurement is an integer in `[0,2^bits)` for some `0 < bits < 64` and the
+/// aggregate is the arithmetic average. This type is a test for demonstrating that the number of measurements
+/// may be used in the `decode_result()` function. No validity checks are done on the input data.
+///
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Average<F: FieldElement> {
+    bits: usize,
+    one: F::Integer,
+}
+
+impl<F: FieldElement> Average<F> {
+    pub fn new(bits: usize) -> Result<Self, FlpError> {
+        let bits_int = F::Integer::try_from(bits).map_err(|err| {
+            FlpError::Encode(format!(
+                "bit length ({}) cannot be represented as a field element: {:?}",
+                bits, err,
+            ))
+        })?;
+
+        if F::modulus() >> bits_int == F::Integer::from(F::zero()) {
+            return Err(FlpError::Encode(format!(
+                "bit length ({}) exceeds field modulus",
+                bits,
+            )));
+        }
+
+        let one = F::Integer::from(F::one());
+
+        Ok(Self { bits, one })
+    }
+}
+
+impl<F: FieldElement> Type for Average<F> {
+    type Measurement = F::Integer;
+    type AggregateResult = f64;
```

Comment:
As I mentioned [here](https://github.com/cfrg/draft-irtf-cfrg-vdaf/pull/95#discussion_r916372075), I'm a bit wary of having the result type (`f64`) have a different precision than the measurements (`u64`, `u128`, ...). Maybe this concern is overblown?

Reply:
I do not quite understand the direction that you would like to go with this. It doesn't seem to be about possible loss of information, as in your alternative information is also lost in the `floor()`. Is this about floats being more "lossy" when dealing with larger numbers?

I think that if this (or sth. similar) is the concern, then it is going to depend on the concrete situation where such an average type is going to be used, and the trade-offs cannot really be decided a priori. Alternatively, it could be that something else is going on here that I don't see?

This is also interesting for our bounded-norm fixed-point vector type because there we currently also use floats. In particular, in our use-case, we are going to have a `float -> fixed-point (prio) -> float` pipeline in some form, simply because ML frameworks use floats.

Reply:
Good point about information loss. And yeah, basically my concern is: When one is given a float type, they expect some imprecision inherent to the IEEE standard for floating point arithmetic. But if the measurements are integers (`u64` or `u128` depending on the field being used to represent them), one would expect to get full precision from the result. Put another way: `Average` is a fairly straightforward extension of `Sum`, but is strictly less useful because it loses precision. 

> This is also interesting for our bounded-norm fixed-point vector type because there we currently also use floats in our aggregate-result-type. In particular, in our use-case, we are going to have a `float -> fixed-point (prio) -> float` pipeline in some form, simply because ML frameworks use floats.

This sounds totally reasonable! One way to solve this might be to have the measurement type for `Average` be `f64` so that the measurement type matches the aggregate result, precisely as you would for your new type.